styx-api-service/src/main/java/com/spotify/styx/api/workflow/WorkflowInitializer.java

```diff
@@ -96,8 +96,8 @@ public WorkflowInitializer(Storage storage, Time time) {
   private TriggerInstantSpec initializeNaturalTrigger(Workflow workflow) {
     final Instant now = time.get();
     final Schedule schedule = workflow.configuration().schedule();
-    final Instant nextTrigger = lastInstant(now, schedule);
-    final Instant nextWithOffset = workflow.configuration().addOffset(nextTrigger);
-    return TriggerInstantSpec.create(nextTrigger, nextWithOffset);
+    final Instant nextTrigger = nextInstant(now, schedule); // Next trigger instant, rounded off by schedule
+    final Instant nextWithOffset = workflow.configuration().subtractOffset(nextTrigger); // gives offset by date
```

Comment:
I think this logic is wrong.

Consider the case where a `hourly` workflow has a offset of `30m`. The `TriggerInstantSpec` would then be created with an `instant` of `HH:30:00` but is should be `HH:00:00`. 

I think maybe what you want to do here is only subtract the offset from `now` and not make any other changes.

```java
// Pretend we're in the past for the purpose of scheduling the first trigger
final Instant now = workflow.configuration().subtractOffset(time.get()); 
```

Reply:
Agreed and then it will not cause the issue mentioned in another comment that it becomes backfill without concurrency control.